src/main/java/org/skriptlang/skript/bukkit/tags/elements/ExprTagsOf.java

```diff
@@ -0,0 +1,129 @@
+package org.skriptlang.skript.bukkit.tags.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Keywords;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.RequiredPlugins;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.Keyed;
+import org.bukkit.Tag;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.tags.TagModule;
+import org.skriptlang.skript.bukkit.tags.TagType;
+import org.skriptlang.skript.bukkit.tags.sources.TagOrigin;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.ThreadLocalRandom;
+
+@Name("Tags of X")
+@Description({
+	"Returns all the tags of an item, block, or entity.",
+	"`minecraft tag` will return only the vanilla tags, `datapack tag` will return only datapack-provided tags, " +
+	"`paper tag` will return only Paper's custom tags (if you are running Paper), " +
+	"and `custom tag` will look in the \"skript\" namespace for custom tags you've registered.",
+	"You can also filter by tag types using \"item\", \"block\", or \"entity\"."
+})
+@Examples({
+	"broadcast minecraft tags of dirt",
+	"send true if paper item tags of target block contains paper tag \"doors\"",
+	"broadcast the block tags of player's tool"
+})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper (paper tags)")
+@Keywords({"blocks", "minecraft tag", "type", "category"})
+public class ExprTagsOf extends PropertyExpression<Object, Tag> {
+
+	static {
+		Skript.registerExpression(ExprTagsOf.class, Tag.class, ExpressionType.PROPERTY,
+				"[all [of]] [the] " + TagOrigin.getFullPattern() + " " + TagType.getFullPattern() + " tags of %itemtype/entity/entitydata%",
```

Comment:
```suggestion
				"[all [[of] the]|the]" + TagOrigin.getFullPattern() + " " + TagType.getFullPattern() + " tags of %itemtype/entity/entitydata%",
```
Since we do not want to allow `all of`